test/foundry/integration/Allo.t.sol

```diff
@@ -1,120 +1,512 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity ^0.8.19;
 
-import {Test, console} from "forge-std/Test.sol";
-import {Allo} from "../../../contracts/core/Allo.sol";
-import {Registry, Metadata} from "../../../contracts/core/Registry.sol";
+import {IAllo} from "contracts/core/interfaces/IAllo.sol";
+import {Metadata, IRegistry} from "contracts/core/Registry.sol";
 import {DonationVotingMerkleDistributionDirectTransferStrategy} from
-    "../../../contracts/strategies/donation-voting-merkle-distribution-direct-transfer/DonationVotingMerkleDistributionDirectTransferStrategy.sol";
+    "contracts/strategies/donation-voting-merkle-distribution-direct-transfer/DonationVotingMerkleDistributionDirectTransferStrategy.sol";
 import {DonationVotingMerkleDistributionBaseStrategy} from
-    "../../../contracts/strategies/donation-voting-merkle-base/DonationVotingMerkleDistributionBaseStrategy.sol";
+    "contracts/strategies/donation-voting-merkle-base/DonationVotingMerkleDistributionBaseStrategy.sol";
 import {ISignatureTransfer} from "permit2/ISignatureTransfer.sol";
 import {IBiconomyForwarder} from "./IBiconomyForwarder.sol";
 import {ECDSA} from "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IntegrationBase} from "./IntegrationBase.sol";
 
-contract IntegrationAllo is Test {
+contract IntegrationAllo is IntegrationBase {
     using ECDSA for bytes32;
 
-    Allo public allo;
-    Registry public registry;
+    IAllo public allo;
     DonationVotingMerkleDistributionDirectTransferStrategy public strategy;
 
-    address public owner;
     address public relayer;
-    address public treasury;
     address public userAddr;
 
+    address public recipient0Addr;
+    address public recipient1Addr;
+    address public recipient2Addr;
```

Comment:
We should move recipients to the IntegrationBase aswell and the token we use in this case dai IMO